src/vs/gitpod/browser/workbench/contrib/exportLogs.contribution.ts

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable code-import-patterns */
+/* eslint-disable header/header */
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IBrowserWorkbenchEnvironmentService } from 'vs/workbench/services/environment/browser/environmentService';
+import { IRemoteAgentService } from 'vs/workbench/services/remote/common/remoteAgentService';
+import { IFileService } from 'vs/platform/files/common/files';
+import { Action2, MenuId, registerAction2 } from 'vs/platform/actions/common/actions';
+import { CATEGORIES } from 'vs/workbench/common/actions';
+import { ServicesAccessor } from 'vs/platform/instantiation/common/instantiation';
+import { URI } from 'vs/base/common/uri';
+import * as resources from 'vs/base/common/resources';
+import { IProgressService, ProgressLocation } from 'vs/platform/progress/common/progress';
+import { CancellationTokenSource } from 'vs/base/common/cancellation';
+import type * as zipModule from '@zip.js/zip.js';
+import { Schemas } from 'vs/base/common/network';
+import { triggerDownload } from 'vs/base/browser/dom';
+
+const getZipModule = (function () {
+	let zip: typeof zipModule;
+	return async () => {
+		if (!zip) {
+			// when actually importing the module change `zip.js` to `zipjs`
+			// without the dot because loader.js will do a check for `.js` extension
+			// and it won't resolve the module path correctly
+			// @ts-ignore
+			zip = await import('@zip.js/zipjs');
+			zip.configure({
+				useWebWorkers: false
+			});
+		}
+		return zip;
+	};
+})();
+
+registerAction2(class ExportLogsAction extends Action2 {
+	constructor() {
+		super({
+			id: 'gitpod.workbench.exportLogs',
+			title: { original: 'Export all logs', value: 'Export all logs' },
+			category: CATEGORIES.Developer,
+			menu: {
+				id: MenuId.CommandPalette
+			}
+		});
+	}
+
+	async run(accessor: ServicesAccessor): Promise<void> {
+		const progressService = accessor.get(IProgressService);
+		const fileService = accessor.get(IFileService);
+		const environmentService = accessor.get(IBrowserWorkbenchEnvironmentService);
+		const remoteAgentService = accessor.get(IRemoteAgentService);
+
+		const cts = new CancellationTokenSource();
+		const bufferData = await progressService.withProgress<Uint8Array | undefined>(
+			{
+				location: ProgressLocation.Dialog,
+				title: 'Exporting logs to zip file ...',
+				cancellable: true,
+				delay: 1000
+			},
+			async progress => {
+				const token = cts.token;
```

Comment:
I wonder why don’t we’ll collect all output logs instead? 

It should be more robust since we don’t need to know concrete paths? And we can get more info like if Java does not work we can ask to use the same command and inspect Java output?